components/org.wso2.carbon.identity.authz.valve/src/main/java/org/wso2/carbon/identity/authz/valve/util/Utils.java

```diff
@@ -43,6 +43,14 @@ public static String getTenantDomainFromURLMapping(Request request) {
                 temp = temp.substring(0, index);
                 domain = temp;
             }
+        } else if (requestURI.contains("/o/")) {
+            // TODO: decide whether to use ->
```

Comment:
can we finalize this?

Reply:
since the domain is set to PrivilegedCarbonContext, modified the code to use that instead of retrieving it as a request attribute.